main.py

```diff
@@ -743,6 +774,168 @@ def complete_match(string: str, matches: Iterable) -> list:
             possible_matches.add(possible)
     return sorted(possible_matches)
 
+
+@bot.command()
+async def randomize(content, message):
+    max_mages = 2
+    max_spells = 4
+    max_gems = 3
+    max_relics = 2
+    min_complexity = 1
+    max_complexity = 11
+    max_set_complexity = 12
+
+    min_rating = 1
+    max_rating = 10
+
+    split_content = content.split()
+    if len(split_content) % 2 < 1:
+        msg = "Wrong argument count arguments are: -p <PlayerCount> \
+-m <GemsRelicsSpells> -minD <minimalNemesisDificulty> -maxD <maximumNemesisDificulty>" \
+
+        log(msg)
+        await message.channel.send(msg)
+        return
+
+    params = {}
+    for i in range(1, len(split_content)-1, 2):
+        params[split_content[i]] = split_content[i+1]
+
+    if "-p" in params:
+        if not isInt(params["-p"]) or not int(params["-p"]) in range(1, 5):
+            msg = "Wrong argument -p must be between 1 and 4 : -p 3"
+            log(msg)
+            await message.channel.send(msg)
+            return
+        max_mages = int(params["-p"])
+
+    if "-m" in params:
+        if not len(params["-m"]) == 3 or not isInt(params["-m"]):
+            msg = "Wrong argument -m must be GRS : -m 324"
+            log(msg)
+            await message.channel.send(msg)
+            return
+        max_gems = int(params["-m"][0])
+        max_relics = int(params["-m"][1])
+        max_spells = int(params["-m"][2])
+
+    if (max_spells + max_gems + max_relics) != 9:
+        msg = "Market wont sum up to 9!"
+        log(msg)
+        await message.channel.send(msg)
+        return
+
+    if "-minD" in params:
+        if not isInt(params["-minD"]) or not int(params["-minD"]) in range(1, 11):
+            msg = "Wrong argument -minD must be between 1 and 10 : -minD 3"
+            log(msg)
+            await message.channel.send(msg)
+            return
+        min_complexity = int(params["-minD"])
+
+    if "-maxD" in params:
+        if not isInt(params["-maxD"]) or not int(params["-maxD"]) in range(1, 11):
+            msg = "Wrong argument -maxD must be between 1 and 10 : -maxD 8"
+            log(msg)
+            await message.channel.send(msg)
+            return
+        max_complexity = int(params["-maxD"])
+
+    if max_complexity < min_complexity:
+        max_complexity = min_complexity+1
+    if min_complexity > max_complexity:
+        min_complexity = max_complexity-1
+    if min_complexity > max_set_complexity:
+        min_complexity = max_set_complexity-1
+
+    #some args should be like: expansion ignored or complexity
+
+   #Random Nemesis
+    msg = "Ok why don't you try out this Challange? \n \n"
```

Comment:
`msg` should be a `List[str]` and do `msg.append(x)` instead of `msg += x`, then `'\n'.join(msg)` when you mean to use it. The newlines shouldn't be a part of the string itself.